### ISSUE TITLE
Cannot run out of the box on bittrex

### ISSUE THREAD

**Issue by hroff-1902** [freqtrade/freqtrade#2880]
Our `config.json.example` is used as a sample config for Bittrex (as `config_bittrex.json.example`, but with other name). And it's used by setup.sh to create new config, with Bittrex as default choice.

If we populate SampleStrategy into user_data or create a new strategy with `new-strategy`, freqtrade will fail:

```
$ freqtrade trade --dry -s SampleStrategy
...
2020-02-07 23:03:08,661 - freqtrade - ERROR - Exchange Bittrex does not support market orders.
```

That's because config.json.example does not redefinr "order_types" and `order_types['stoploss'] == 'market'`. 

And this is with all default options (default choices in setup.sh, and with SampleStrategy) chosen by a new user, which produces bad user experience.

Possible solutions:
* redefine `order_types['stoploss']` in config.json.example -- bad, because it's supposed to be a generic config example;
* same, but rename config.json.example into config_bittrex.json.example. In this case, setup.sh should be changed to fetch proper config depending on the exchange name selected by the user.
* redefine order_types['stoploss'] in the template for new strategy and in the sample_strategy.py -- bad, since it's also supposed to be generic
* change default choice in setup.sh from bittrex to binance -- pretty well for me, but this should be synchronized with the rest of code, so that binance become the default exchange...

There is no good and quick solution for this... :(

**Comment by xmatthias** [comment]
Well, if you look at #2831 you'll see that it contains the solution (well at least for the most part).

It removes the "question" part from setup.sh - and moves that into freqtrade code.
Since that's not a shell script anymore, it allows us to run this on windows too (and provides more flexibility as well).
This allows to select the exchange - not prefering any exchange (they should be sorted alphabetically), and a few other things.
i've intentionally kept it simple for now - but future improvements could include an "advanced" section where many more questions are asked.

Now it won't fix this issue 100% - as the "stoploss" problem still remains, however in combination with moving "stoploss" to a limit order in the sample strategy  (which makes sense - on binance you "should" use stoploss-on-exchange - so the "stoploss" key does not matter) it should work out of the box again.

**Comment by xmatthias** [comment]
do we consider this as fixed by #2831 and #2890?

**Comment by hroff-1902** [comment]
not yet, I'm preparing a PR which would close it.
(the complete order_types dict should be added into the config template for Bittrex)
